tests/convert_secret_extract_to_static_extract/secret_extracts.mlir

```diff
@@ -0,0 +1,26 @@
+// RUN: heir-opt --convert-secret-extract-to-static-extract %s | FileCheck %s
+
+// CHECK-LABEL: @extract_at_secret_index
+func.func @extract_at_secret_index(%arg0: !secret.secret<tensor<32xi16>>, %arg1: !secret.secret<index>) -> !secret.secret<i16> {
+    %c0_i16 = arith.constant 0 : i16
+    %0 = secret.generic ins(%arg0, %arg1 : !secret.secret<tensor<32xi16>>, !secret.secret<index>) {
+    ^bb0(%arg2: tensor<32xi16>, %arg3: index):
+      // CHECK: %[[FOR:.*]] = affine.for %[[I:.*]] = 0 to 32
+      // CHECK-NEXT: %[[INNER_FOR:.*]] = affine.for %[[J:.*]] = 0 to 32
+      // CHECK-NEXT: %[[COND:.*]] = arith.cmpi eq, %[[J]], %[[SECRET_INDEX:.*]] : index
+      // CHECK-NEXT: %[[EXTRACTED:.*]] = tensor.extract %[[TENSOR:.*]][%[[J]]]
+      // CHECK: %[[IF:.*]] = scf.if %[[COND:.*]] -> (i16) {
+      // CHECK-NEXT:   scf.yield %[[EXTRACTED]] : i16
+      // CHECK-NEXT: } else {
+      // CHECK-NEXT:   scf.yield %[[DUMMY_VAL:.*]] : i16
+      %1 = affine.for %i = 0 to 32 iter_args(%sum = %c0_i16) -> (i16) {
+        %extracted = tensor.extract %arg2[%arg3] {size = 32} : tensor<32xi16>
+        %extracted_0 = tensor.extract %arg2[%i] {size = 32} : tensor<32xi16>
```

Comment:
I noticed one of your tests has a loop in the input (extract) and one doesn't (insert). I guess seeing that it handles the nested loop case is nice, so I'd suggest having one non-loop and one loop test case for each test file.